### PR TITLE
Code Model: When setting the name of a CodeElement, ensure that its members are updated

### PR DESCRIPTION
**Pull request by DustinCampbell**
**Customer Scenario**: This bug can manifest in many ways as its an issue in the Code Model implementation, but the scenario reported is as follows:

1. Create a V1 Workflow Sequential Console Application project (targeting .NET Framework 3.5).
2. Click on the Workflow Designer
3. In the Properties window, change the (Name) of the workflow to something else (e.g. "NewWorkflow")

This results in a very unhelpful dialog:

image

When the user clicks OK, the (Name) property gets reverted to what it was before. However, the name was actually partially changed by the time the exception was thrown. So, the user finds that several files have been modified but the process was incomplete.

**Fix Description**: Code Model tracks nodes with a "node key" concept, which is effectively the full name of an element coupled with an ordinal. When the name of a ```CodeElement``` is modified view ```SetName```. its node key must also be updated since the name of the underlying node has changed. However, we had failed to update the node keys of any members of that ```CodeElement```, which means that they can no longer be located after their parent container's name changes. This is now fixed by using the same ```NodeKeysValidation``` helper that ```CodeElement.RenameSymbol``` uses.

**Testing Done**: Manually verified customer scenario working properly, and added unit test that fails without the fix and passes with the fix. Additionally, the Workflow team has been given private binaries with the fix to run their automation against.

Pinging @jasonmalinowski, @Pilchie, @pharring as stellar reviewers.

### CODE REVIEW

**Comment by pharring** [comment]
:+1:

File: src/VisualStudio/Core/Impl/CodeModel/InternalElements/AbstractKeyedCodeElement.cs [hunk 1]
```diff
@@ -106,7 +107,14 @@ protected override string GetName()
 
         protected override void SetName(string value)
         {
-            UpdateNodeAndReaquireNodeKey(FileCodeModel.UpdateName, value);
+            var nodeKeyValidation = new NodeKeyValidation();
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Impl/CodeModel/InternalElements/AbstractKeyedCodeElement.cs]: Does this need the EnsureEditor() that UpdateNodeAndReacquireNodeKey was doing?

Comment (DustinCampbell) [inline src/VisualStudio/Core/Impl/CodeModel/InternalElements/AbstractKeyedCodeElement.cs]: Great question! The answer is, yes! It's always better to be paranoid that the invisible editor is created.

**Comment by MattGertz** [comment]
Approved 6/2 by ML Shiproom.

**Comment by DustinCampbell** [comment]
@dotnet-bot please test this

**Comment by DustinCampbell** [comment]
Jenkins, retest this please.

**Comment by Pilchie** [comment]
Good to be merged now?

**Pull request merged**